pytorch_lightning/utilities/device_dtype_mixin.py

```diff
@@ -194,4 +189,8 @@ def apply_fn(module):
             if dtype is not None:
                 module._dtype = dtype
 
+        # update LightningModule device
+        self._device = device
```

Comment:
it's not necessary, as I see it `self.apply` already runs the apply_fn to itself.
https://github.com/pytorch/pytorch/blob/0a72904ab4a2eef45fab33b3b7ed79a9ebfa2d60/torch/nn/modules/module.py#L617
Correct?

Furthermore, there would probably be a bug here because device can be None!
If we do `self.half()` this would set the device to None! Careful

Reply:
thanks, I didn't know that. Yes, it works without it as expected.